datasets/qa/nctc_sanctions/crawler.py

```diff
@@ -1,37 +1,31 @@
 import json
 from typing import List
 from rigour.mime.types import JSON
-from prefixdate import parse_formats
 
 from zavod import Context
 from zavod import helpers as h
 
 TYPES = {"1": "Person", "2": "Organization"}
-FORMATS = ["%Y-%m-%d", "%Y-%m-%d-%H", "%d/%m/%Y", "X_%Y_X_X", "%d-%m-%Y", "%Y"]
 ALIAS_SPLITS = [";", "original script", "(", ")", "previously listed as"]
 
 
-def parse_dates(text: str) -> List[str]:
+def parse_dates(context: Context, text: str) -> List[str]:
     if not len(text):
         return []
     dates = set()
     type_, text = text.split("_", 1)
     if text == "X_X_X_X":
         return []
-    for part in h.multi_split(text, [":", ";", "؛", " to "]):
-        part = part.replace("___", " ")
-        part = part.replace("_X_X_X", " ")
+    for part in h.multi_split(
+        text,
+        [":", ";", "؛", " to ", "حوالي", "_", "والي"],
+    ):
         part = part.strip()
         if part == "00":
             continue
-        parsed = parse_formats(part, FORMATS)
-        if parsed.text is not None:
-            dates.add(parsed.text)
-        else:
-            dates.update(h.extract_years(part))
-    # if not len(dates):
-    #     print(text)
-    # return [text]
+        parsed_dates = h.extract_date(context.dataset, part)
```

Comment:
Just as an idea for a pattern, this function could also return a list of unparsed dates that have been split and pre-cleaned, and then send those into `h.apply_date`.